source/wifi/wifi_hal.c

```diff
@@ -9203,28 +9203,26 @@ INT wifi_createVAP(wifi_radio_index_t index, wifi_vap_info_map_t *map)
     return RETURN_OK;
 }
 
-int parse_channel_list_int_arr(int radioIndex, char *pchannels, wifi_channels_list_t* chlistptr)
+int parse_channel_list_int_arr(char *pchannels, wifi_channels_list_t* chlistptr)
 {
-
-    char *token;
+    char *token, *next;
     const char s[2] = ",";
     int count =0;
 
     /* get the first token */
-    token = strtok(pchannels, s);
+    token = strtok_r(pchannels, s, &next);
 
     /* walk through other tokens */
-    while( token != NULL ) {
-        chlistptr->channels_list[count] = atoi(token);
-        count++;
-        token = strtok(NULL, s);
+    while( token != NULL && count < MAX_CHANNELS) {
+        chlistptr->channels_list[count++] = atoi(token);
+        token = strtok_r(next, s, &next);
```

Comment:
On the first call to strtok_r(), str should point to the string to be parsed, and the value of saveptr is ignored. In subsequent calls, str should be NULL, and saveptr should be unchanged since the previous call. 
---> strtok_r(NULL, s, &next)

Reply:
Not sure about MAX_CHANNELS, the shitproof version
sizeof(array) / sizeof(array[0])

Reply:
Hi Tomasz,
MAX_CHANNELS is defined in wifi_hal_generic.h

#define MAX_CHANNELS    64

And, i think i have address your review comments regarding strtok_r.

Reply:
thanks for approving the changes, @tomasz-kilarski-plume .